File: 0387/main.ts
function firstUniqChar(s: string): number {
    let hashmap: Map<string, number> = new Map()
    let index = 0
    let value = ""
    let len = s.length

    for(index = 0; index < len; index++){
        value = s[index]
        if (hashmap.get(value) != undefined){
            hashmap.set(value, -1)
        } else {
            hashmap.set(value, index)
        }
    }

    let minIndex = 1e9

    hashmap.forEach((index) => {
        if (index != -1 && index < minIndex){
            minIndex = index
        }
    })

    if (minIndex == 1e9){
        return -1
    }

    return minIndex
};

File: 0387/main.go
package main

func firstUniqChar(s string) int {
	hashmap := make(map[rune]int)

	var index int
	var value rune
	var ok bool
	for index, value = range s {
		if _, ok = hashmap[value]; ok {
			hashmap[value] = -1
		} else {
			hashmap[value] = index
		}
	}

	var minIndex int = 1e9
	for value, index = range hashmap {
		if index != -1 && index < minIndex {
			minIndex = index
		}
	}
	if minIndex == 1e9 {
		return -1
	}
	return minIndex
}
